tasks/src/lib.rs

```diff
@@ -342,6 +355,42 @@ pub enum SchedulerError {
     /// It's better to halt the execution to avoid damage.
     #[error("unexpected tag in command completion result")]
     UnexpectedTaggedResponse(Tagged<'static>),
+    #[error("unexpected BYE response")]
+    UnexpectedByeResponse(Bye<'static>),
+}
+
+pub struct TaskRunner<'a, T: Task> {
+    scheduler: &'a mut Scheduler,
+    handle: TaskHandle<T>,
+}
+
+impl<T: Task> Flow for TaskRunner<'_, T> {
+    type Event = T::Output;
+    type Error = SchedulerError;
+
+    fn enqueue_input(&mut self, bytes: &[u8]) {
+        self.scheduler.enqueue_input(bytes);
+    }
+
+    fn progress(&mut self) -> Result<Self::Event, FlowInterrupt<Self::Error>> {
+        loop {
+            match self.scheduler.progress()? {
+                SchedulerEvent::TaskFinished(mut token) => {
+                    if let Some(output) = self.handle.resolve(&mut token) {
+                        break Ok(output);
+                    }
```

Comment:
In the long run we should find a better idea how to deal with unexpected tokens here. But for now, what do you think about addding a log statement with WARN level? `imap-flow` already has `tracing` as logging dependency.